build.sbt

```diff
@@ -32,14 +32,31 @@ inThisBuild(
 
 lazy val common: Seq[Setting[_]] =
   Seq(
-    crossScalaVersions := Seq(Dependencies.Scala213, Dependencies.Scala212),
-    scalaVersion := Dependencies.Scala213,
+    crossScalaVersions := Dependencies.CrossScalaVersions,
+    scalaVersion := Dependencies.CrossScalaVersions.head,
     crossVersion := CrossVersion.binary,
     scalafmtOnCompile := true,
     sonatypeProfileName := "com.lightbend",
+    headerLicense := Some(HeaderLicense.Custom("""Copyright (C) 2022 Lightbend Inc. <https://www.lightbend.com>""")),
     // Setting javac options in common allows IntelliJ IDEA to import them automatically
     Compile / javacOptions ++= Seq("-encoding", "UTF-8", "-source", "1.8", "-target", "1.8"),
-    headerLicense := Some(HeaderLicense.Custom("""Copyright (C) 2022 Lightbend Inc. <https://www.lightbend.com>""")),
+    scalacOptions ++= {
+      var scalacOptionsBase = Seq("-encoding", "UTF-8", "-feature", "-unchecked", "-deprecation")
+      if (scalaVersion.value == Dependencies.Scala212)
+        scalacOptionsBase ++: Seq("-Xfuture", "-Xfatal-warnings", "-Xlint", "-Ywarn-dead-code")
+      else if (scalaVersion.value == Dependencies.Scala213)
+        scalacOptionsBase ++: Seq("-Xfatal-warnings", "-Xlint", "-Ywarn-dead-code", "-Wconf:cat=deprecation:info")
+      else
+        scalacOptionsBase
+    },
+    javacOptions ++= (
+      if (isJdk8) Seq.empty
+      else Seq("--release", "8")
```

Comment:
Should this be the other way around? if `isJdk8` then `Seq("--release", "8")`?

Reply:
We always want to build the release jars so that they can be used with JDK 8. The --release flag means that you can use JDK 11 or 17 but anyway release for 8.

I'm not sure why it's not set when using JDK 8 in the build. Maybe the flag doesn't exist in JDK 8. This is what we had in akka-management, so I used the same. Anyway, if building with JDK 8 the jars will be for 8.